uniqueUsernameGenerator fixes and enhancements

LiorKovalio opened the pull request:
Multiple changes:
* Simplified uniqueUsernameGenerator length and added tests
* (BREAKING) Fixed a bug in randomNumber, missing the last stretch (e.g. [91-99]). Simplified randomNumber
* ~Added new styles camelCase and pascalCase~
* ~(BREAKING) Changed uniqueUsernameGenerator default behaviour to filter all dictionaries out of entries containing the separator. If a dictionary turns empty, throw an error.~
* ~Removed a faulty test case. The test wrongly assumed that the default dictionaries contain no dash '-'~

LiorKovalio committed:
Fixed a bug in randomNumber, missing the last stretch (e.g. [91-99]). Simplified randomNumber

LiorKovalio committed:
Simplified uniqueUsernameGenerator length and added tests

subhamg commented:
Hi @LiorKovalio  — thanks for the contribution, and apologies for the delay in getting back to you!

It looks like this PR now has some merge conflicts with main. Could you please resolve those so we can review and move forward?

Thanks again for your patience and for improving the project! 🙌

LiorKovalio commented:
> Hi @LiorKovalio — thanks for the contribution, and apologies for the delay in getting back to you!
> 
> It looks like this PR now has some merge conflicts with main. Could you please resolve those so we can review and move forward?
> 
> Thanks again for your patience and for improving the project! 🙌

Rebased and fixed.
3 of my commits became obsolete after the recent changes in the repo (separator is handled better, camalCase and pascalCase styles were added), so I dropped them and crossed the changes out in the original PR message.
What's left in this PR is the randomNumber bugfix and the maxLength simplification.

Thank you for maintaining this project 🙏
